File: MoSmartPark/MoSmartPark.Services/Database/Notification.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace MoSmartPark.Services.Database
{
    public class Notification
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        [MaxLength(100)]
        public string NotificationType { get; set; } = string.Empty;

        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = string.Empty;

        [Required]
        [MaxLength(1000)]
        public string Message { get; set; } = string.Empty;

        [MaxLength(50)]
        public string? RelatedEntityType { get; set; }

        public int? RelatedEntityId { get; set; }

        public bool IsRead { get; set; } = false;

        public bool IsSent { get; set; } = true;

        public DateTime? SentAt { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation property
        public User User { get; set; } = null!;
    }
}

File: MoSmartPark/UI/mosmartpark_desktop/lib/model/notification.dart
import 'package:json_annotation/json_annotation.dart';

part 'notification.g.dart';

@JsonSerializable()
class Notification {
  final int id;
  final int userId;
  final String notificationType;
  final String title;
  final String message;
  final String? relatedEntityType;
  final int? relatedEntityId;
  final bool isRead;
  final bool isSent;
  final DateTime? sentAt;
  final DateTime createdAt;
  final String? userName;

  Notification({
    this.id = 0,
    this.userId = 0,
    this.notificationType = '',
    this.title = '',
    this.message = '',
    this.relatedEntityType,
    this.relatedEntityId,
    this.isRead = false,
    this.isSent = true,
    this.sentAt,
    required this.createdAt,
    this.userName,
  });

  factory Notification.fromJson(Map<String, dynamic> json) =>
      _$NotificationFromJson(json);
  Map<String, dynamic> toJson() => _$NotificationToJson(this);
}
